music_assistant/controllers/player_queues.py

```diff
@@ -1040,6 +1041,23 @@ async def _load_item(
                 and queue_item.media_item.album.item_id == next_item.media_item.album.item_id
             )
         )
+        current_index = self._queues[queue_id].current_index
```

Comment:
This is a bit dangerous because maybe some new playback started and the queue was first cleared ?
Also this assumes that current_index is already updated (this relies on the player).
I would suggest to just do prev_index = next_index-1 if next_index else None and then ofc guard on the None

Reply:
If I understood correctly then `next_index` is `None` for the last item in the queue so if I get `prev_index` from that then this won't work in this edge case.

I've now made a small change to re-use the `queue` object from line 1017. Would that be safer?

Reply:
It's more that you should not rely on queue.current_item in this part of the code (where its async loading some item into the buffer of a player). Maybe the player is buffering ahead or the item load happens before the metadata is updated. That is why we pass along the item in these functions, because it is the item which is actually being loaded/requested and not what the metadata report is saying we're playing.

Reply:
Thanks for the explanation.